src/components/Table/GridTable.test.tsx

```diff
@@ -1123,11 +1123,11 @@ describe("GridTable", () => {
   });
 });
 
-function Collapse() {
-  const { isCollapsed, toggleCollapse } = useContext(GridCollapseContext);
-  const icon = isCollapsed ? "+" : "-";
+function Collapse({ id }: { id: string }) {
+  const { isCollapsed, toggleCollapsed } = useContext(GridCollapseContext);
+  const icon = isCollapsed(id) ? "+" : "-";
   return (
-    <div onClick={toggleCollapse} data-testid="collapse">
+    <div onClick={() => toggleCollapsed(id)} data-testid="collapse">
```

Comment:
**question:** Should the `onClick` be stored in a `useCallback`? There isn't a whole lot to this component, so probably fine if it re-renders, but figured I'd point it out.

Reply:
I did use a `useCallback` in the `real` use case of this method (see `CollapseToggle` component). Didn't seem worth it in the test 🤷

Reply:
I didn't even realize I was lookign at the test at first. hah.